framework/include/utils/GeometryUtils.h

```diff
@@ -0,0 +1,156 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Moose.h"
+#include "libmesh/point.h"
+#include "libmesh/mesh_tools.h"
+
+namespace geom_utils
+{
+/**
+ * Check whether a point is equal to zero
+ * @param[in] pt point
+ * @return whether point is equal to the zero point
+ */
+bool isPointZero(const Point & pt);
+
+/**
+ * Get the unit vector for a point parameter
+ * @param[in] pt point
+ * @param[in] name name of the parameter
+ */
+Point unitVector(const Point & pt, const std::string & name);
+
+/**
+ * Rotate point about an axis
+ * @param[in] p point
+ * @param[in] angle angle to rotate (radians)
+ * @param[in] axis axis expressed as vector
+ * @return rotated point
+ */
+Point rotatePointAboutAxis(const Point & p, const Real & angle, const Point & axis);
+
+/**
+ * Get the minimum distance from a point to another set of points, in the plane
+ * perpendicular to the specified axis
+ * @param[in] pt point
+ * @param[in] candidates set of points we will find the nearest of
+ * @param[in] axis axis perpendicular to the plane of the polygon
+ * @return minimum distance to the provided points
+ */
+Real minDistanceToPoints(const Point & pt,
+                         const std::vector<Point> & candidates,
+                         const unsigned int & axis);
```

Comment:
I think we will want an enum here instead of unsigned int

Reply:
it's actually kind of nice to be able to acess points directly at the axis. Probably better for performance too. So I m happy with this unsigned int

Reply:
```suggestion
                         const unsigned int axis);
```